src/lib.rs

```diff
@@ -1,54 +1,76 @@
+use std::sync::Once;
+
+#[cfg(feature = "tracing")]
+use tracing::Level;
+
+#[macro_use]
+mod logger;
+
 mod config;
 mod datachannel;
 mod error;
 mod peerconnection;
 mod track;
 
+static INIT_LOGGING: Once = Once::new();
+
 mod sys {
     use std::ffi::CStr;
     use std::os::raw::c_char;
 
     use datachannel_sys as sys;
-    use lazy_static::lazy_static;
-
-    lazy_static! {
-        pub(super) static ref INIT_LOGGING: () = {
-            let level = match log::max_level() {
-                log::LevelFilter::Off => sys::rtcLogLevel_RTC_LOG_NONE,
-                log::LevelFilter::Error => sys::rtcLogLevel_RTC_LOG_ERROR,
-                log::LevelFilter::Warn => sys::rtcLogLevel_RTC_LOG_WARNING,
-                log::LevelFilter::Info => sys::rtcLogLevel_RTC_LOG_INFO,
-                log::LevelFilter::Debug => sys::rtcLogLevel_RTC_LOG_DEBUG,
-                log::LevelFilter::Trace => sys::rtcLogLevel_RTC_LOG_VERBOSE,
-            };
-            unsafe { sys::rtcInitLogger(level, Some(log_callback)) };
-        };
-    }
 
-    unsafe extern "C" fn log_callback(level: sys::rtcLogLevel, message: *const c_char) {
+    pub unsafe extern "C" fn log_callback(level: sys::rtcLogLevel, message: *const c_char) {
```

Comment:
This should stay private, is there a reason to make it `pub` ?

Reply:
This is because I removed `lazy_static` and used the built-in `Once` struct. The ensure_logging/configure logging now use that to initialize the log, w/o having a pub `lazy_static` variable in the sys. I can put lazy_static back, but I think it's not necessary to include an entire crate for a built-in feature.

Reply:
Sure, removing `lazy_static` and using your solution is fine. If you want this function to be available in the crate only then `pub(crate)` should do, making it `pub` exposes it to the public API, which is not required right ?

Reply:
Actually you would need to do `pub mod`, `sys` is completely hidden from the public api :)